pkg/utils/utils.go

```diff
@@ -187,17 +187,16 @@ func GetModulesFromNode(nodeName string, nodeOc *interactive.Oc) []string {
 	return strings.Split(strings.ReplaceAll(output, "\r\n", "\n"), "\n")
 }
 
+// ModuleInTree returns true if the module hasn't tainted the kernel with the
+// out-of-tree ("O") bit. The /sys/module/<module>/taint file has the stringified
+// values of all the taints it is adding to the kernel. Each bit is one letter.
+// Refs:
+//  1. https://www.kernel.org/doc/html/latest/admin-guide/tainted-kernels.html
+//  2. https://github.com/torvalds/linux/blob/master/kernel/panic.c#L369
 func ModuleInTree(nodeName, moduleName string, nodeOc *interactive.Oc) bool {
-	command := `chroot /host modinfo ` + moduleName + ` | awk '{ print $1 }'`
+	command := `chroot /host cat /sys/module/` + moduleName + `/taint`
 	cmdOutput := RunCommandInNode(nodeName, nodeOc, command, timeoutPid)
-	outputSlice := strings.Split(strings.ReplaceAll(cmdOutput, "\r\n", "\n"), "\n")
-	// The output, if found, should look something like 'intree:   Y'.
-	// As long as we look for 'intree:' being contained in the string we should be good to go.
-	found := false
-	if StringInSlice(outputSlice, `intree:`, false) {
-		found = true
-	}
-	return found
+	return !strings.Contains(cmdOutput, "O")
```

Comment:
Is the Taint OOT superseding the other 2 module taints below? If not we could add decoding fro these
[ TAINT_PROPRIETARY_MODULE ]	= { 'P', 'G', true },
[ TAINT_OOT_MODULE ]		= { 'O', ' ', true },
[ TAINT_UNSIGNED_MODULE ]	= { 'E', ' ', true },